Index: code-server/lib/vscode/src/vs/workbench/services/clipboard/browser/clipboardService.ts
===================================================================
--- code-server.orig/lib/vscode/src/vs/workbench/services/clipboard/browser/clipboardService.ts
+++ code-server/lib/vscode/src/vs/workbench/services/clipboard/browser/clipboardService.ts
@@ -3,14 +3,14 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { localize } from '../../../../nls.js';
+// import { localize } from '../../../../nls.js';
 import { InstantiationType, registerSingleton } from '../../../../platform/instantiation/common/extensions.js';
 import { IClipboardService } from '../../../../platform/clipboard/common/clipboardService.js';
 import { BrowserClipboardService as BaseBrowserClipboardService } from '../../../../platform/clipboard/browser/clipboardService.js';
-import { INotificationService, Severity } from '../../../../platform/notification/common/notification.js';
-import { IOpenerService } from '../../../../platform/opener/common/opener.js';
-import { Event } from '../../../../base/common/event.js';
-import { DisposableStore } from '../../../../base/common/lifecycle.js';
+// import { INotificationService, Severity } from '../../../../platform/notification/common/notification.js';
+// import { IOpenerService } from '../../../../platform/opener/common/opener.js';
+// import { Event } from '../../../../base/common/event.js';
+// import { DisposableStore } from '../../../../base/common/lifecycle.js';
 import { IWorkbenchEnvironmentService } from '../../environment/common/environmentService.js';
 import { ILogService } from '../../../../platform/log/common/log.js';
 import { ILayoutService } from '../../../../platform/layout/browser/layoutService.js';
@@ -19,8 +19,8 @@ import { getActiveWindow } from '../../.
 export class BrowserClipboardService extends BaseBrowserClipboardService {
 
 	constructor(
-		@INotificationService private readonly notificationService: INotificationService,
-		@IOpenerService private readonly openerService: IOpenerService,
+		// @INotificationService private readonly notificationService: INotificationService,
+		// @IOpenerService private readonly openerService: IOpenerService,
 		@IWorkbenchEnvironmentService private readonly environmentService: IWorkbenchEnvironmentService,
 		@ILogService logService: ILogService,
 		@ILayoutService layoutService: ILayoutService
@@ -39,6 +39,7 @@ export class BrowserClipboardService ext
 
 	override async readText(type?: string): Promise<string> {
 		this.logService.trace('BrowserClipboardService#readText called with type:', type);
+		// console.log('BrowserClipboardService#readText called with type:', type);
 		if (!!this.environmentService.extensionTestsLocationURI && typeof type !== 'string') {
 			type = 'vscode-tests'; // force in-memory clipboard for tests to avoid permission issues
 		}
@@ -50,34 +51,37 @@ export class BrowserClipboardService ext
 
 		try {
 			const readText = await getActiveWindow().navigator.clipboard.readText();
+			// console.log('BrowserClipboardService#readText with readText.length:', readText);
 			this.logService.trace('BrowserClipboardService#readText with readText.length:', readText.length);
 			return readText;
 		} catch (error) {
-			return new Promise<string>(resolve => {
+			// return new Promise<string>(resolve => {
 
-				// Inform user about permissions problem (https://github.com/microsoft/vscode/issues/112089)
-				const listener = new DisposableStore();
-				const handle = this.notificationService.prompt(
-					Severity.Error,
-					localize('clipboardError', "Unable to read from the browser's clipboard. Please make sure you have granted access for this website to read from the clipboard."),
-					[{
-						label: localize('retry', "Retry"),
-						run: async () => {
-							listener.dispose();
-							resolve(await this.readText(type));
-						}
-					}, {
-						label: localize('learnMore', "Learn More"),
-						run: () => this.openerService.open('https://go.microsoft.com/fwlink/?linkid=2151362')
-					}],
-					{
-						sticky: true
-					}
-				);
-
-				// Always resolve the promise once the notification closes
-				listener.add(Event.once(handle.onDidClose)(() => resolve('')));
-			});
+			// 	// Inform user about permissions problem (https://github.com/microsoft/vscode/issues/112089)
+			// 	const listener = new DisposableStore();
+			// 	const handle = this.notificationService.prompt(
+			// 		Severity.Error,
+			// 		localize('clipboardError', "Unable to read from the browser's clipboard. Please make sure you have granted access for this website to read from the clipboard."),
+			// 		[{
+			// 			label: localize('retry', "Retry"),
+			// 			run: async () => {
+			// 				listener.dispose();
+			// 				resolve(await this.readText(type));
+			// 			}
+			// 		}, {
+			// 			label: localize('learnMore', "Learn More"),
+			// 			run: () => this.openerService.open('https://go.microsoft.com/fwlink/?linkid=2151362')
+			// 		}],
+			// 		{
+			// 			sticky: true
+			// 		}
+			// 	);
+
+			// 	// Always resolve the promise once the notification closes
+			// 	listener.add(Event.once(handle.onDidClose)(() => resolve('')));
+			// });
+			this.logService.error(error);
+			return '';
 		}
 	}
 }
Index: code-server/lib/vscode/src/vs/workbench/contrib/outline/browser/outline.contribution.ts
===================================================================
--- code-server.orig/lib/vscode/src/vs/workbench/contrib/outline/browser/outline.contribution.ts
+++ code-server/lib/vscode/src/vs/workbench/contrib/outline/browser/outline.contribution.ts
@@ -3,17 +3,17 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { localize, localize2 } from '../../../../nls.js';
-import { IViewsRegistry, Extensions as ViewExtensions } from '../../../common/views.js';
-import { OutlinePane } from './outlinePane.js';
+import { localize } from '../../../../nls.js';
+// import { IViewsRegistry, Extensions as ViewExtensions } from '../../../common/views.js';
+// import { OutlinePane } from './outlinePane.js';
 import { Registry } from '../../../../platform/registry/common/platform.js';
 import { IConfigurationRegistry, Extensions as ConfigurationExtensions, ConfigurationScope } from '../../../../platform/configuration/common/configurationRegistry.js';
-import { VIEW_CONTAINER } from '../../files/browser/explorerViewlet.js';
-import { SyncDescriptor } from '../../../../platform/instantiation/common/descriptors.js';
-import { Codicon } from '../../../../base/common/codicons.js';
-import { registerIcon } from '../../../../platform/theme/common/iconRegistry.js';
+// import { VIEW_CONTAINER } from '../../files/browser/explorerViewlet.js';
+// import { SyncDescriptor } from '../../../../platform/instantiation/common/descriptors.js';
+// import { Codicon } from '../../../../base/common/codicons.js';
+// import { registerIcon } from '../../../../platform/theme/common/iconRegistry.js';
 import { OutlineConfigKeys } from '../../../services/outline/browser/outline.js';
-import { IOutlinePane } from './outline.js';
+// import { IOutlinePane } from './outline.js';
 
 // --- actions
 
@@ -21,21 +21,21 @@ import './outlineActions.js';
 
 // --- view
 
-const outlineViewIcon = registerIcon('outline-view-icon', Codicon.symbolClass, localize('outlineViewIcon', 'View icon of the outline view.'));
-
-Registry.as<IViewsRegistry>(ViewExtensions.ViewsRegistry).registerViews([{
-	id: IOutlinePane.Id,
-	name: localize2('name', "Outline"),
-	containerIcon: outlineViewIcon,
-	ctorDescriptor: new SyncDescriptor(OutlinePane),
-	canToggleVisibility: true,
-	canMoveView: true,
-	hideByDefault: false,
-	collapsed: true,
-	order: 2,
-	weight: 30,
-	focusCommand: { id: 'outline.focus' }
-}], VIEW_CONTAINER);
+//const outlineViewIcon = registerIcon('outline-view-icon', Codicon.symbolClass, localize('outlineViewIcon', 'View icon of the outline view.'));
+//禁用outline图标，避免与文件资源管理器图标冲突
+// Registry.as<IViewsRegistry>(ViewExtensions.ViewsRegistry).registerViews([{
+// 	id: IOutlinePane.Id,
+// 	name: localize2('name', "Outline"),
+// 	containerIcon: outlineViewIcon,
+// 	ctorDescriptor: new SyncDescriptor(OutlinePane),
+// 	canToggleVisibility: true,
+// 	canMoveView: true,
+// 	hideByDefault: false,
+// 	collapsed: true,
+// 	order: 2,
+// 	weight: 30,
+// 	focusCommand: { id: 'outline.focus' }
+// }], VIEW_CONTAINER);
 
 // --- configurations
 
Index: code-server/lib/vscode/src/vs/workbench/contrib/timeline/browser/timeline.contribution.ts
===================================================================
--- code-server.orig/lib/vscode/src/vs/workbench/contrib/timeline/browser/timeline.contribution.ts
+++ code-server/lib/vscode/src/vs/workbench/contrib/timeline/browser/timeline.contribution.ts
@@ -6,8 +6,8 @@
 import { localize } from '../../../../nls.js';
 import { SyncDescriptor } from '../../../../platform/instantiation/common/descriptors.js';
 import { Registry } from '../../../../platform/registry/common/platform.js';
-import { IViewsRegistry, IViewDescriptor, Extensions as ViewExtensions } from '../../../common/views.js';
-import { VIEW_CONTAINER } from '../../files/browser/explorerViewlet.js';
+import { IViewDescriptor } from '../../../common/views.js';
+// import { VIEW_CONTAINER } from '../../files/browser/explorerViewlet.js';
 import { ITimelineService, TimelinePaneId } from '../common/timeline.js';
 import { TimelineHasProviderContext } from '../common/timelineService.js';
 import { TimelinePane } from './timelinePane.js';
@@ -62,7 +62,7 @@ configurationRegistry.registerConfigurat
 	}
 });
 
-Registry.as<IViewsRegistry>(ViewExtensions.ViewsRegistry).registerViews([new TimelinePaneDescriptor()], VIEW_CONTAINER);
+// Registry.as<IViewsRegistry>(ViewExtensions.ViewsRegistry).registerViews([new TimelinePaneDescriptor()], VIEW_CONTAINER);
 
 namespace OpenTimelineAction {
 
